mailet/standard/src/test/java/org/apache/james/transport/mailets/MailAttributesListToMimeHeadersTest.java

```diff
@@ -0,0 +1,132 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+
+package org.apache.james.transport.mailets;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import javax.mail.MessagingException;
+
+import com.google.common.collect.ImmutableList;
+import jdk.nashorn.internal.ir.annotations.Immutable;
+import org.apache.mailet.Mailet;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.apache.mailet.base.test.MailUtil;
+import org.apache.mailet.base.test.MimeMessageBuilder;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class MailAttributesListToMimeHeadersTest {
+
+    @Rule public ExpectedException expectedException = ExpectedException.none();
+
+    private Mailet mailet;
+
+    private final String HEADER_NAME1 = "JUNIT";
+    private final String HEADER_NAME2 = "JUNIT2";
+
+    private final ImmutableList<String> MAIL_ATTRIBUTE_VALUE1 = ImmutableList.of("test1.1", "test1.2");
+    private final ImmutableList<String> MAIL_ATTRIBUTE_VALUE2 = ImmutableList.of("test2.1", "test2.2");
+
+    private final String MAIL_ATTRIBUTE_NAME1 = "org.apache.james.test";
+    private final String MAIL_ATTRIBUTE_NAME2 = "org.apache.james.test2";
+
+    @Before
+    public void setup() {
+        mailet = new MailAttributesListToMimeHeaders();
+    }
+
+    @Test
+    public void shouldPutAttributesIntoHeadersWhenMappingDefined() throws MessagingException {
+        FakeMailetConfig mailetConfig = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("simplemapping",
+                        MAIL_ATTRIBUTE_NAME1 + "; " + HEADER_NAME1 +
+                        "," + MAIL_ATTRIBUTE_NAME2 + "; " + HEADER_NAME2 +
+                        "," + "another.attribute" + "; " + "Another-Header")
+                .build();
+        mailet.init(mailetConfig);
+
+        FakeMail mockedMail = MailUtil.createMockMail2Recipients(MailUtil.createMimeMessage());
+        mockedMail.setAttribute(MAIL_ATTRIBUTE_NAME1, MAIL_ATTRIBUTE_VALUE1);
+        mockedMail.setAttribute(MAIL_ATTRIBUTE_NAME2, MAIL_ATTRIBUTE_VALUE2);
+        mockedMail.setAttribute("unmatched.attribute", "value");
+
+        mailet.service(mockedMail);
+
+        assertThat(mockedMail.getMessage().getHeader(HEADER_NAME1)).containsAll(MAIL_ATTRIBUTE_VALUE1);
+        assertThat(mockedMail.getMessage().getHeader(HEADER_NAME1)).hasSameSizeAs(MAIL_ATTRIBUTE_VALUE1);
```

Comment:
:+1: 
But I had to introduce new variable just to get array. It's a bit frustrating